### PR TITLE
fix enjoy vizdoom error

### PR DESCRIPTION
**Pull request by wmFrank**

### CODE REVIEW

**Comment by alex-petrenko** [comment]
Should we fix the tests first before I merge it?

**Comment by wmFrank** [comment]
> Should we fix the tests first before I merge it?

Absolutely I should fix it before merging. We must use the newest gym version, otherwise mujoco will not work. Instead of rendering, newest gym uses pygame, so I am working on changing the vizdoom render function to pygame.

**Comment by wmFrank** [comment]
Yeah, I think it is ready to be merged.

File: sample_factory/envs/doom/doom_gym.py [hunk 1]
```diff
@@ -441,17 +442,26 @@ def render(self, mode="human"):
...
+            import pygame
+
+            if self.screen is None:
+                pygame.init()
+                pygame.display.init()
+                self.screen = pygame.display.set_mode((render_w, render_h))
+
+            self.surf = pygame.surfarray.make_surface(np.rot90(img))
+            self.screen.blit(self.surf, (0, 0))
+            pygame.display.update()
```
Comment (alex-petrenko) [inline sample_factory/envs/doom/doom_gym.py:462]: Ok, good solution!

File: sample_factory/envs/doom/doom_gym.py [hunk 2]
```diff
@@ -463,8 +473,13 @@ def close(self):
         except RuntimeError as exc:
             log.warning("Runtime error in VizDoom game close(): %r", exc)
 
-        if self.viewer is not None:
-            self.viewer.close()
+        # if self.viewer is not None:
+        #     self.viewer.close()
```
Comment (alex-petrenko) [inline sample_factory/envs/doom/doom_gym.py:477]: Unless it serves a purpose, try not to leave commented code like this. To find old code we can always use git :)

File: setup.py [hunk 3]
```diff
@@ -47,8 +47,8 @@
     extras_require={
         'dev': ['black', 'isort', 'pytest<8.0', 'flake8'],
         'atari': ['gym[atari, accept-rom-license]'],
-        'mujoco': ['gym[mujoco]<1.0', 'mujoco', 'mujoco_py>=2.0,<3.0'],
-        'vizdoom': ['vizdoom<2.0'],
+        'mujoco': ['gym[mujoco, mujoco_py]'],
+        'vizdoom': ['vizdoom<2.0', 'gym[classic_control]'],
```
Comment (alex-petrenko) [inline setup.py:51]: I'm assuming this is for pygame dependency? Cool.

**Review by alex-petrenko — APPROVED** [review]

File: setup.py (hunk 3 above)
Comment (wmFrank) [reply setup.py:51]: Yes, it is for pygame

**Pull request merged**